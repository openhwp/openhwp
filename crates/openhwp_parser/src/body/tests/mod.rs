use super::{hash_code, pseudo};
use crate::u32;

#[test]
fn officially_distributed_hwp_5_0_format() {
    static PAYLOAD: [u8; 256] = [
        190, 100, 247, 91, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 159, 159, 168, 159, 219,
        159, 168, 159, 170, 159, 217, 236, 213, 236, 169, 236, 218, 236, 175, 236, 213, 236, 221,
        236, 222, 236, 157, 175, 195, 134, 194, 134, 199, 134, 195, 134, 194, 76, 122, 76, 117, 76,
        123, 76, 146, 165, 227, 165, 231, 165, 156, 165, 150, 165, 149, 165, 227, 165, 203, 249,
        204, 249, 207, 249, 193, 249, 201, 249, 193, 22, 83, 22, 46, 22, 37, 19, 38, 19, 36, 19,
        82, 19, 16, 147, 19, 19, 19, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47, 47,
        190, 190, 190, 190, 190, 190, 190, 190, 98, 98, 31, 31, 31, 31, 18, 18, 18, 18, 18, 18, 18,
        18, 18, 204, 204, 204, 204, 204, 204, 204, 204, 204, 204, 204, 204, 204, 164, 164, 164,
        164, 164, 164, 164, 164, 164, 164, 164, 164, 164, 164, 128, 128, 128, 128, 128, 128, 128,
        128, 128, 128, 128, 128, 128, 128, 128, 128, 218, 218, 218, 218, 218, 218, 20, 20, 20, 20,
        20, 20, 20, 20, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 170, 170, 170, 170,
        170, 170, 170, 152, 152, 152, 152, 44, 44, 44, 44, 44, 44, 44, 204, 204, 204, 204, 204,
        204, 204, 204, 204, 245, 245, 245, 245, 245, 245, 245, 245, 245, 245, 245, 245, 245, 245,
        245, 245, 97,
    ];
    let seed = u32(&PAYLOAD, 0);
    let pseudo = pseudo(seed);
    let hash_code = hash_code(seed, pseudo, PAYLOAD);
    insta::assert_debug_snapshot!((seed, pseudo, hash_code));
}
