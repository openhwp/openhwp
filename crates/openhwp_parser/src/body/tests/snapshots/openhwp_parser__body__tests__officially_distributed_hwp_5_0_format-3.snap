---
source: crates/openhwp_parser/src/body/tests/mod.rs
expression: hash_code
---
[
    55,
    0,
    68,
    0,
    55,
    0,
    53,
    0,
    53,
    0,
    57,
    0,
    69,
    0,
    54,
    0,
]
